libgadget/density.c

```diff
@@ -450,14 +479,20 @@ density_postprocess(int i, TreeWalk * tw)
 {
     if(P[i].Type == 0)
     {
-        if(SPHP(i).Density <= 0)
-            endrun(12, "Particle %d has bad density: %g\n", i, SPHP(i).Density);
         int PI = P[i].PI;
         MyFloat * DhsmlDens;
         if(DENSITY_GET_PRIV(tw)->DoEgyDensity)
             DhsmlDens = &(DENSITY_GET_PRIV(tw)->DhsmlDensityFactor[PI]);
         else
             DhsmlDens = &(SPHP(i).DhsmlEgyDensityFactor);
+        if(SPHP(i).Density <= 0) {
+            if(P[i].NumNgb == 0) {
+                SPHP(i).Density = 1;
```

Comment:
Applying the MINHSML bound?

Reply:
There is almost no particle nearby, but in certain 'symmetriziation' of SPH pairs the density may contribute to other particles. I suspect 0 here may be safter than a undimensioned number 1.

Reply:
Done - this almost puts back the original condition from gadget-2. The MINHSML bound should not affect this - if NumNgb == 0, the code will increase Hsml.